packages/edit-site/src/components/block-editor/editor-canvas.js

```diff
@@ -40,9 +54,30 @@ function EditorCanvas( { enableResizing, settings, children, ...props } ) {
 			style={ enableResizing ? {} : deviceStyles }
 			ref={ mouseMoveTypingRef }
 			name="editor-canvas"
-			className="edit-site-visual-editor__editor-canvas"
+			className={ classnames( 'edit-site-visual-editor__editor-canvas', {
+				'is-focused': isFocused && canvasMode === 'view',
+			} ) }
 			{ ...props }
 			role={ canvasMode === 'view' ? 'button' : undefined }
+			tabIndex={ canvasMode === 'view' ? 0 : undefined }
```

Comment:
All of these `canvasMode==='view'` things give me the heebie jeebies. Is there a way to avoid that?

Reply:
Almost every prop relies on `canvasMode==='view'`. Maybe it would be cleaner to just move the conditional out and have two instance of the <IFrame> component?

Reply:
Good point! I refactored it again to [hopefully make it even easier to read](https://github.com/WordPress/gutenberg/pull/50296/commits/39a9db6ca8650cabd3efd6634b92b178aaf165e6). Lmk what you think!